### PR TITLE
Add TBLIS

### PR DESCRIPTION
**Pull request by chillenb**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by chillenb** [comment]
@conda-forge/help-c-cpp, ready for review!

File: recipes/tblis/build.sh [hunk 1]
```diff
@@ -0,0 +1,7 @@
+CFLAGS=$(echo "${CFLAGS}" | sed "s/-march=[a-zA-Z0-9]*//g")
```
Comment (xhochy) [inline recipes/tblis/build.sh:1]: Why are you removing this?

File: recipes/tblis/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+  run_exports:
+    - {{ pin_subpackage(name, min_pin='x.x', max_pin='x.x') }}
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ stdlib("c") }}
+    - {{ compiler('cxx') }}
+    - make
+    - libhwloc
```
Comment (xhochy) [inline recipes/tblis/meta.yaml:24]: This should probably only be in host
```suggestion
```

File: recipes/tblis/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/devinamatthews/tblis/archive/4de1919.tar.gz
+  sha256: 85b72884022edd2612e3a0b3ed12aa6237d3989b581091d21f58124a7450aaeb
+
+build:
+  skip: true  # [not (unix and x86_64)]
```
Comment (xhochy) [inline recipes/tblis/meta.yaml:13]:
```suggestion
  skip: true  # [win]
```

**Review by xhochy — CHANGES_REQUESTED** [review]

File: recipes/tblis/build.sh (hunk 1 above)
Comment (chillenb) [reply recipes/tblis/build.sh:1]: TBLIS builds optimized microkernels for a wide range of x86 processors and needs to pass in specific -march flags to compile properly. The default conda-forge compile flags `-march=nocona -mtune=haswell` will interfere with this.

TBLIS detects the user's architecture at runtime and selects the best codepath. Therefore, no illegal instruction errors should occur. The conda-forge recipe for blis [also uses this line](https://github.com/conda-forge/blis-feedstock/blob/main/recipe/build.sh).

File: recipes/tblis/meta.yaml (hunk 2 above)
Comment (chillenb) [reply recipes/tblis/meta.yaml:24]: ok!

**Review by xhochy — APPROVED** [review]

**Pull request merged**